### PR TITLE
added a docker file

### PR DESCRIPTION
**Pull request by nikolawhallon**
Added a docker file.

### CODE REVIEW

File: Dockerfile
Comment (mlodato517) [inline Dockerfile]: I don't know anything about anything docker related but I'd rather not use a multi-year-old version in our public example. I'm guessing you used this one because of [the link you supplied above](https://blog.ubuntu.com/2018/07/09/minimal-ubuntu-released) indicating that 18.04 was a minimal image. I believe that is true and at the time of the blog post (9 July 2018) it probably was the first such minimal release. Looking at [the official ubuntu docker repo](https://hub.docker.com/_/ubuntu) it shows all sorts of tags as supported (including 22.04 which is LTS) and  the `Locales` section mentions:
> Given that it is a minimal install of Ubuntu

and it's not referencing a particular version. So I think the 22.04 Ubuntu image is _also_ a minimal install and we should use that and we could delete that comment above. Could be wrong though! Would happily get a second opinion :laughing:

File: Dockerfile
Comment (mlodato517) [inline Dockerfile:3]: We should come up with some `maintainer="Deepgram <engineering@deepgram.com>"` sometime

File: Dockerfile
Comment (mlodato517) [inline Dockerfile]: [Cliff's PR](https://github.com/deepgram/dgtools/pull/1/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R15) sets this environment variable in a separate step so it doesn't have to be repeated here. Not necessary but maybe cleaner if it works?:
```
ENV DEBIAN_FRONTEND=noninteractive
```

**Review by mlodato517 — COMMENTED** [review]
I'm like 99% sure this is good but I'm not super smart about these. Might want to bring in someone who knows something about Docker :speak_no_evil:

**Comment by nikolawhallon** [comment]
> I'm like 99% sure this is good but I'm not super smart about these. Might want to bring in someone who knows something about Docker speak_no_evil

I don't know anything about Dockerfiles, so someone else's input would be great.

**Review by jcdyer — COMMENTED** [review]
I'd agree with using a more modern docker image.  

Not important, but you could also bump the rust-toolchain to 1.61 now.

**Comment by nikolawhallon** [comment]
I'll see if Ubuntu 22.04 even works.

**Comment by nikolawhallon** [comment]
Ubuntu 22.04 does not work:

```
E: Unable to locate package libssl1.0.0
E: Couldn't find any package by glob 'libssl1.0.0'
E: Couldn't find any package by regex 'libssl1.0.0'
```
is there a working example somewhere? I am still on Ubuntu 20.04 myself, and I can, indeed, install `libssl-dev`.

**Comment by mlodato517** [comment]
Looks like maybe [it's version 3.0.2 now](https://packages.ubuntu.com/search?keywords=libssl-dev&searchon=names&suite=jammy&section=all). Ubuntu 22.04 probably just doesn't have the ancient version :laughing:

**Comment by nikolawhallon** [comment]
> Looks like maybe [it's version 3.0.2 now](https://packages.ubuntu.com/search?keywords=libssl-dev&searchon=names&suite=jammy&section=all). Ubuntu 22.04 probably just doesn't have the ancient version laughing

that link gave me a 5xx. I can try.

**Comment by nikolawhallon** [comment]
@mlodato517 nope:
```
E: Unable to locate package libssl3.0.2
E: Couldn't find any package by glob 'libssl3.0.2'
E: Couldn't find any package by regex 'libssl3.0.2'
```

**Comment by mlodato517** [comment]
```
sudo apt install libssl3
```
looks like it would work for me if it wasn't already installed.

**Comment by nikolawhallon** [comment]
very slow to test these things via `docker build`, hold on

**Comment by nikolawhallon** [comment]
Seems to have worked - ubuntu image and rust version have been updated.

File: Dockerfile
Comment (nikolawhallon) [reply Dockerfile]: how's this: 9cb2e05e2e7766448890abc9396be355b85ed03f

File: Dockerfile
Comment (mlodato517) [reply Dockerfile:3]: It looks like, for now, we don't have that, so maybe we'll come back and do that in The Future. Here's [a Slack thread with maybe more details](https://deepgram.slack.com/archives/GA7Q4BH2M/p1654113176111789).

**Review by mlodato517 — APPROVED** [review]
It looks like the only comments where Ubuntu and Rust version and those were updated so, with my continued ignorance, I approve this to make triple sure we don't hold up the blog post.

**Pull request merged**
